dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/validator/relationship/DataRelationsValidator.java

```diff
@@ -63,25 +62,26 @@ private void validateRelationshipReference( Reporter reporter, TrackerBundle bun
         RelationshipItem item )
     {
         Optional<String> uid = getUidFromRelationshipItem( item );
-        TrackerType trackerType = relationshipItemValueType( item );
+        String trackerType = relationshipItemValueType( item );
```

Comment:
If we agree that we can make assumptions, can we have the uid as a simple String? If this validator only run after another validator already checked that the uid is there, we can assume that

Reply:
I think it's a good idea to assume the uid will always be present here. I'll move the LinkValidator so it's run before the DataRelationsValidator, this way we also assure there's only one uid per relationship item.